java/arcs/android/storage/database/AndroidSqliteDatabaseManager.kt

```diff
@@ -126,6 +126,29 @@ class AndroidSqliteDatabaseManager(
         db.runGarbageCollection()
     }
 
+    override suspend fun getEntitiesCount(inMemory: Boolean): Long {
+        return registry
+            .fetchAll()
+            .filter { it.isPersistent == !inMemory }
+            .map { getDatabase(it.name, it.isPersistent).getEntitiesCount() }
+            .fold(0L) { sum, element -> sum + element }
+    }
+
+    override suspend fun getStorageSize(inMemory: Boolean): Long {
+        return registry
+            .fetchAll()
+            .filter { it.isPersistent == !inMemory }
+            .map { getDatabase(it.name, it.isPersistent).getSize() }
+            .fold(0L) { sum, element -> sum + element }
+    }
+
+    override suspend fun isStorageTooLarge(): Boolean {
+        return registry
+            .fetchAll()
+            .map { databaseSizeTooLarge(getDatabase(it.name, it.isPersistent)) }
+            .fold(false) { sum, element -> sum || element }
```

Comment:
can you use
.filter { databaseSizeTooLarge(getDatabase(it.name, it.isPersistent)) }
.any()

Reply:
sure, changed.